src/app/api/agencies/route.ts

```diff
@@ -12,7 +17,27 @@ export async function GET() {
     }));
     return Response.json(filteredAgencies);
   } catch (error) {
-    console.error(error);
+    return new Response('Internal Server Error', { status: 500 });
+  }
+}
+
+export async function POST(request: Request) {
+  try {
+    const body = await request.json();
+    const { services, ...agency } = body;
+
+    const serviceIds = [];
+    for (const service of services) {
+      const createdService = await createService(service);
+      serviceIds.push(createdService._id);
+    }
+
+    await createAgency({ ...agency, services: serviceIds });
+    return new Response('Success', { status: 200 });
+  } catch (error) {
+    if (error instanceof ApiError) {
+      return new Response(error.message, { status: error.statusCode });
+    }
     return new Response('Internal Server Error', { status: 500 });
```

Comment:
I think it might be beneficial to start using something like the [JSend specification](https://github.com/omniti-labs/jsend) to standardize responses from our api. What do you think?

Reply:
Sounds good. I will get the implemented right away.